tests/Unit/Evolution/DiscontinuousGalerkin/Actions/Test_NormalCovectorAndMagnitude.cpp

```diff
@@ -0,0 +1,306 @@
+                                        domain::Tags::Mesh<Dim - 1>>>(*box)
+            .at(direction)
+            .number_of_grid_points()};
+    if constexpr (not UseFlatSpace) {
+      double temp = 0.1;
+      for (auto& component : get<InverseSpatialMetric<Dim>>(fields_on_face)) {
+        for (double& t : component) {
+          t = temp;
+          temp *= 1.1;
+        }
+      }
+    }
+
+    db::mutate<
+        evolution::dg::Tags::InternalFace::NormalCovectorAndMagnitude<Dim>>(
+        box,
+        &evolution::dg::Actions::detail::
+            unit_normal_vector_and_covector_and_magnitude<
+                tmpl::conditional_t<UseFlatSpace, FlatSpaceSystem,
+                                    SystemWithInverseMetric<Dim>>,
+                Dim, field_face_tags>,
+        make_not_null(&fields_on_face), direction,
+        unnormalized_normal_covectors,
+        db::get<domain::CoordinateMaps::Tags::CoordinateMap<Dim, Frame::Grid,
+                                                            Frame::Inertial>>(
+            *box));
+
+    if constexpr (UseFlatSpace) {
+      CHECK_ITERABLE_APPROX(
+          (DataVector{fields_on_face.number_of_grid_points(), 1.0}),
+          get(magnitude(
+              get<evolution::dg::Tags::InternalFace::NormalCovector<Dim>>(
+                  *get<evolution::dg::Tags::InternalFace::
+                           NormalCovectorAndMagnitude<Dim>>(*box)
+                       .at(direction)))));
+    } else {
+      const auto& normal_covector =
+          get<evolution::dg::Tags::InternalFace::NormalCovector<Dim>>(
+              *get<evolution::dg::Tags::InternalFace::
+                       NormalCovectorAndMagnitude<Dim>>(*box)
+                   .at(direction));
+      CHECK_ITERABLE_APPROX(
+          (DataVector{fields_on_face.number_of_grid_points(), 1.0}),
+          sqrt(get(
+              dot_product(normal_covector, normal_covector,
+                          get<InverseSpatialMetric<Dim>>(fields_on_face)))));
+    }
+
+    for (const double magnitude_of_normal :
+         get(get<evolution::dg::Tags::InternalFace::MagnitudeOfNormal>(
+             *get<evolution::dg::Tags::InternalFace::NormalCovectorAndMagnitude<
+                  Dim>>(*box)
+                  .at(direction)))) {
+      CHECK(magnitude_of_normal > 0.0);
+    }
+    for (size_t i = 0; i < Dim; ++i) {
+      const auto& normal_covector_and_magnitude =
+          *get<evolution::dg::Tags::InternalFace::NormalCovectorAndMagnitude<
+              Dim>>(*box)
+               .at(direction);
+      CHECK_ITERABLE_APPROX(
+          unnormalized_normal_covectors.at(direction).get(i),
+          DataVector{
+              get<evolution::dg::Tags::InternalFace::NormalCovector<Dim>>(
+                  normal_covector_and_magnitude)
+                  .get(i) *
+              get(get<evolution::dg::Tags::InternalFace::MagnitudeOfNormal>(
+                  normal_covector_and_magnitude))});
```

Comment:
I would find this easier to read if you assigned the lengthy RHS to a helpfully-named variable, perhaps `const DataVector unit_normal_times_magnitude = ...`, so that the CHECK_ITERABLE_APPROX itself became easier to parse.